### PR TITLE
11 get reviews queries

### PR DESCRIPTION
**Pull request by lilyha1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -220,3 +220,33 @@ describe("5. GET /api/users", () => {
...
+describe("6. GET /api/reviews (queries)", () => {
+      test("status:200, responds with indicated category (query)", () => {
+    return request(app)
+      .get("/api/reviews?category=dexterity")
+      .expect(200)
+      .then(({ body }) => {
+        const { reviews } = body;
+        expect(reviews).toBeInstanceOf(Array);
+        expect(reviews).toHaveLength(1);
+        expect(reviews).toBeSortedBy("created_at", { descending: true });
```
Comment (dav3rid) [inline __tests__/app.test.js]: It seems a little odd to assert a sort order on an array you've asserted only has a single element in - can remove if you'd like

File: __tests__/app.test.js [hunk 2]
```diff
@@ -220,3 +220,33 @@ describe("5. GET /api/users", () => {
...
+        reviews.forEach((reviews) => {
+          expect(reviews).toEqual(
+            expect.objectContaining({
+              owner: expect.any(String),
+              title: expect.any(String),
+              review_id: expect.any(Number),
+              category: expect.any(String),
```
Comment (dav3rid) [inline __tests__/app.test.js]: This property does want to be specific here - nothing about this assertion proves reviews from the correct category are present

File: __tests__/app.test.js [hunk 3]
```diff
@@ -248,5 +248,103 @@ describe("6. GET /api/reviews (queries)", () => {
...
+              review_img_url: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              designer: expect.any(String),
+              comment_count: expect.any(Number),
+            })
+          );
+        });
+      });
+  });
```
Comment (dav3rid) [inline __tests__/app.test.js:443]: Makes sense to have changed the category - still need to assert the category property though

File: __tests__/app.test.js [hunk 4]
```diff
@@ -248,5 +248,103 @@ describe("6. GET /api/reviews (queries)", () => {
...
+  test("status:200, sorts by asc)", () => {
+    return request(app)
+      .get("/api/reviews?sort_by=title&order=ASC")
+      .expect(200)
+      .then(({ body }) => {
+        const { reviews } = body;
+        expect(reviews).toBeInstanceOf(Array);
+        expect(reviews).toHaveLength(13);
+        expect(reviews).toBeSortedBy("title", { ascending: true });
+        reviews.forEach((reviews) => {
```
Comment (dav3rid) [inline __tests__/app.test.js]: No real need to assert the structure of each object in the array here - you've already done that in previous tests - just the sort order and direction would be sufficient

File: __tests__/app.test.js [hunk 5]
```diff
@@ -248,5 +248,103 @@ describe("6. GET /api/reviews (queries)", () => {
...
+  });
+
+  test("status:400, invalid sort query", () => {
+    return request(app)
+      .get("/api/reviews?sort_by=invalid")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid sort query");
+      });
+  });
```
Comment (dav3rid) [inline __tests__/app.test.js:491]: error tests look good but need some more - same problems but for order
Another consideration - what if a category exists but has no reviews associated?

File: db/models/models.js [hunk 6]
```diff
@@ -1,26 +1,49 @@
...
-      GROUP BY users.username, reviews.review_id
-      ORDER BY reviews.created_at DESC;
-        `
-    )
-    .then((results) => {
-      return results.rows;
-    });
+      `;
+  if (category != undefined) {
+    queryStr += `WHERE reviews.category = '${category}'`;
```
Comment (dav3rid) [inline db/models/models.js]: You're open to sql injection here! MUST use parameterised query $ syntax instead

File: db/models/models.js [hunk 7]
```diff
@@ -1,26 +1,49 @@
...
+  }
+
+  queryStr += ` 
+ GROUP BY reviews.owner, reviews.title, reviews.review_id
+ ORDER BY ${sort_by} ${order};`;
```
Comment (dav3rid) [inline db/models/models.js:49]: Also here - order will need to be manually sanitised above just like sort_by

File: db/models/models.js [hunk 8]
```diff
@@ -1,26 +1,49 @@
...
+  return db.query(queryStr).then((results) => {
+    const reviews = results.rows;
+    if (reviews.length === 0) {
```
Comment (dav3rid) [inline db/models/models.js:53]: This is where you could ensure exactly why the array is empty

**Review by dav3rid — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js [hunk 9]
```diff
@@ -285,66 +269,52 @@ describe("6. GET /api/reviews (queries)", () => {
...
-            })
-          );
-        });
+test("status:200, category and sorts by asc)", () => {
+  return request(app)
+    .get("/api/reviews?category=social deduction&sort_by=title&order=ASC")
+    .expect(200)
+    .then(({ body }) => {
+      const { reviews } = body;
+      expect(reviews).toBeInstanceOf(Array);
```
Comment (MKD01) [inline __tests__/app.test.js]: The assertion of this being an array does not ned to be made as you have a forEach which is an array method itself, so would make that check for you.

**Review by MKD01 — APPROVED** [review]
Happy for this to be merged after these changes have been made

**Pull request merged**